examples/access-control-migration/README.md

```diff
@@ -1,6 +1,306 @@
+to involve other users as team members. The system has an administration.
+
+Here is the diagram that describes the models:
+
+![models](./imgs/scenario-model.png)
+
+And the Project endpoints with their allowed roles:
+
+- /Projects/list-projects
+  - everyone
+- /Projects/view-all
+  - admin
+- /Projects/{id}/show-balance
+  - owner, teamMember
+- /Projects/{id}/donate
+  - admin, owner, teamMember
+- /Projects/{id}/withdraw
+  - owner
+
+The diagram for models also include the pre-created data in this application.
+
+For example, u1 owns project1, it also creates a team with u1 and u2 as members.
+Which means u1 is the **owner** of project1, and u1 and u2 are the **team
+members** of project1. And u3 is the **admin**.
+
+# Difference
+
+LoopBack 3 has several built-in models that consists of a RBAC system. The
+models are User, Role, RoleMapping, ACL, AccessToken. You can learn how they
+work together in tutorial
+[Controlling data access](https://loopback.io/doc/en/lb3/Controlling-data-access.html).
+
+LoopBack 4 authorization system gives developers the flexibility to implement
+the RBAC on their own. You can leverage popular 3rd-party libraries like
+[casbin](https://github.com/casbin/casbin) or [oauth0](https://auth0.com/) for
+the role mapping.
+
+This guide includes a demo for using [casbin](#steps-with-casbin) and using
+[oauth0](#steps-with-oauth0)(TBD)
+
+# Steps with Casbin
+
+Next let's migrate the LoopBack 3 access control example to LoopBack 4. Here is
+an overview of the steps:
+
+- Create models (migrates model properties)
+- Set up login endpoint (migrate User endpoint)
+- Set up project endpoints (migrate Project endpoints)
+- Set up authentication (migrate boot/authentication.js)
+- Set up authorization (This is the core of the tutorial, migrate role resolvers
+  and model acls)
+- Seed data (migrate boot/sample-models.js)
+
+## Create models
+
+There are 4 models involved in this example. 3 of them are migrated from
+[the original models](https://github.com/strongloop/loopback-example-access-control/tree/master/common/models)
+(Project, Team, User). And we add one more model UserCredentials to separate the
+sensitive information from the User model.
+
+You can run `lb4 model` to create the 4 models, and run `lb4 repository` to
+create their corresponding persistency layer.
+
+_LoopBack 3 model provides 3 layers: data shape, persistency, REST APIs, while
+LoopBack 4 model only describes the data shape. Therefore in this section we
+also need to create repositories for data persistency. You can read about their
+difference in document
+[migrating model definitions and built-in APIs](https://loopback.io/doc/en/lb4/migration-models-core.html)_
+
+_To keep the tutorial concise and focus on the core implementation, the detailed
+commands are list in the reference [model creation](link_tbd)._
+
+## Set up login endpoint
+
+LoopBack 3 has a default User model with a bunch of pre-defined APIs exposed. In
```

Comment:
User model -> `User` model ( or User model -> User model `User`)
User controller -> User controller `Usercontroller`

Reply:
hmm, same as https://github.com/strongloop/loopback-next/pull/4706#discussion_r385274056, IMO `UserController` refers to the concrete class. 
I'd like to use it as a concept here, no code involved.